File: 2021/10/01/BinarySearch.java
/**
 * https://leetcode.com/problems/binary-search/
 * @author Andrew Jarombek
 * @since 10/1/2021
 */

public class BinarySearch {
    public int search(int[] nums, int target) {
        int low = 0;
        int high = nums.length;
        int mid = high / 2;

        while (low < high) {
            int val = nums[mid];
            if (target < val) {
                high = mid;
            } else if (val == target) {
                return mid;
            } else {
                low = mid + 1;
            }

            mid = (high + low) / 2;
        }

        return -1;
    }
}

File: 2022/04/24/binarySearch.ts
/**
 * https://leetcode.com/problems/binary-search/
 * @author Andrew Jarombek
 * @since 4/24/2022
 */

function search(nums: number[], target: number): number {
    let low = 0;
    let high = nums.length;

    while (low <= high) {
        const mid = Math.floor((low + high) / 2);
        const val = nums[mid];

        if (val == target) {
            return mid;
        } else if (val > target) {
            high = mid - 1;
        } else {
            low = mid + 1;
        }
    }

    return -1;
}